File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrong

import "math"

// IsNumber returns true is the given integer is an Armstrong number.
func IsNumber(n int) bool {
	numberLen := math.Floor(math.Log10(float64(n))) + 1
	sum := 0
	orig := n

	for n > 0 {
		sum += int(math.Pow(float64(n%10), float64(numberLen)))
		n /= 10
	}

	return sum == orig
}

File: solutions/lua/armstrong-numbers/1/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
    local sum = 0
    local n = number
    local digits = math.floor(math.log(number, 10)) + 1

    while n > 0 do
        sum = sum + (n % 10) ^ digits
        n = math.floor(n / 10)
    end

    return number == sum
end

return ArmstrongNumbers
